project/SparkBuild.scala

```diff
@@ -765,16 +750,6 @@ object CopyDependencies {
 
 }
 
-object Java8TestSettings {
-  import BuildCommons._
-
-  lazy val settings = Seq(
-    javacJVMVersion := "1.8",
-    // Targeting Java 8 bytecode is only supported in Scala 2.11.4 and higher:
-    scalacJVMVersion := (if (System.getProperty("scala-2.10") == "true") "1.7" else "1.8")
```

Comment:
I see, so we need scalac to target 1.7 byte code for the Scala code it emits in 2.10. That would be fine for moving to Java 8 if javac (8) were handling Java code using Java 8, because the result would just be a mix of 7 and 8 bytecode but no big deal. Do I have that right?